src/api/app/models/repository.rb

```diff
@@ -166,12 +166,30 @@ def is_kiwi_type?
     name == "images"
   end
 
+  def has_local_path?
+    path_elements.each do |pe|
+      return true if pe.link.project == project
+    end
+
+    false
+  end
+
   def clone_repository_from(source_repository)
     source_repository.repository_architectures.each do |ra|
       repository_architectures.create architecture: ra.architecture, position: ra.position
     end
 
-    if source_repository.is_kiwi_type?
+    position = 1
+    if source_repository.has_local_path?
+      # we need to refer to the local variant of it instead which has to exist already
```

Comment:
don't link to the original external repo, but use the repo from this project pointing to this external repo.